### PR TITLE
added the webapp for ffp

### LINKED ISSUE #10: Create a Simple WebApp for Forest Fire Prediction 👨‍💻
Hey, @Nibba2018 and @BALaka-18 I have created this issue to give a direction to this small project.

Now, we have already done the Data analysis and Model building using the forest fire dataset.
In data analysis, we have covered Uni-variate, Bi-variate, and Multi-variate analysis of the data.
In the model Building part, we have made a Regression model using Random Forest Regressor.

Now, if we can **give a direction to all the analysis and model building parts** that we have done by making a small **web app** for forest fire prediction using basic flask or streamlet and any beginner-friendly hosting platform(_Heroku or Ngrok or Netlify_) that will help this project to complete the whole **life cycle of a Data Science project**.

Let me know it is acceptable or not.

### PR DESCRIPTION
**Pull request by soumya997**
# Description
Included the files related to the web-app and its deployment.

Fixes #10 

# Added Files:
1.app.py
2. Procfile
3.requirments.txt
4.setup.sh
5. ffp2_model.pkl

# Web-app link:
https://ffp-web-app.herokuapp.com/

### CODE REVIEW

**Comment by soumya997** [comment]
This command at the 5th step [`git push origin --set-upstream origin <branch-name>`](https://github.com/dsc-iem/AI-Hacktoberfest/issues/10#issuecomment-717031968)  was giving an error.
<br>
<a href="https://ibb.co/v4RzWpP">Screenshot-642</a>

So I had to try this `git push -u origin <branch-name>` 👇 I hope it is ok.
<br>
<a href="https://ibb.co/t3pPWjW">Screenshot-643</a><br />

File: app.py
Comment (Nibba2018) [inline app.py]: Make the exception handling a bit more robust. `Please fill up all the inputs` will still show even if the user has entered all the fields. Add different message for different Exceptions without printing the entire stacktrace.

File: app.py
Comment (Nibba2018) [inline app.py]: Add some placeholder texts to let the user know what kind of values the model is expecting. Take information from [here](https://github.com/dsc-iem/AI-Hacktoberfest/blob/master/README.md).

Also it will be great to have some field hints or info explaining what these fields mean.

**Review by Nibba2018 — CHANGES_REQUESTED** [review]
Thanks for the PR @soumya997 , I have added some comments below kindly address those and after that it will ready to be merged.

File: app.py
Comment (Nibba2018) [reply app.py]: this is what I am talking about: 
image

File: app.py
Comment (Nibba2018) [reply app.py]: @soumya997 Also make sure to add the deployed link in README

**Comment by Nibba2018** [comment]
@soumya997 Another request I have is to sort the contributors in descending order based on their number of commits to this project. You can refer [this](https://github.com/fury-gl/fury/blob/master/docs/source/ext/github_tools.py) to have an idea how to implement it.

File: app.py
Comment (soumya997) [reply app.py]: Actually, all the fields accept only numeric value. so, here `Jan` must be a number.
Yes, I can write the instruction to fill the inputs with only numeric values.

**Comment by soumya997** [comment]
> @soumya997 Another request I have is to sort the contributors in descending order based on their number of commits to this project. You can refer [this](https://github.com/fury-gl/fury/blob/master/docs/source/ext/github_tools.py) to have an idea how to implement it.


Ok,let me try.
Do I have to install `fury` for that?

**Comment by Nibba2018** [comment]
@soumya997 No no, that's just the code to get the information about contributors using the github API

File: app.py
Comment (Nibba2018) [reply app.py]: Yeah that's I am asking you to make the exceptions more clearer to understand.

**Comment by soumya997** [comment]
> @soumya997 Another request I have is to sort the contributors in descending order based on their number of commits to this project. You can refer [this](https://github.com/fury-gl/fury/blob/master/docs/source/ext/github_tools.py) to have an idea of how to implement it.

It will give the no. of commits, each contributor has done, right? and based on that I have to arrange the GitHub card in the `contributors page`, right?
If yes then we can simply see the no. of commits from [here](https://github.com/dsc-iem/AI-Hacktoberfest/graphs/contributors) and arrange the cards accordingly.

**Comment by Nibba2018** [comment]
@soumya997 Yeah. Use whatever implementation you feel like.

**Comment by soumya997** [comment]
> @soumya997 Yeah. Use whatever implementation you feel like.

I have changed the ordering based on no. of commits and no of lines added(I took the `no.of lines added parameter` because some people are having the same number of commits).

**Comment by soumya997** [comment]
> @soumya997 Another request I have is to sort the contributors in descending order based on their number of commits to this project. You can refer [this](https://github.com/fury-gl/fury/blob/master/docs/source/ext/github_tools.py) to have an idea how to implement it.

This is how the contributor page looking rn,
<br>


plz give some response if its ok.

File: app.py
Comment (soumya997) [reply app.py]: I have fixed the exception with a proper massage. So, now if anybody enters any non-numeric value or doesn't enter any value in the inputs then they will see a message.

File: app.py
Comment (soumya997) [reply app.py]: in the st.text_input() I have added some placeholder text(which are basically some instructions on how to fill up the inputs) like this 👇, and also edited the readme with the deployed link.
<be>

I hope it will do the job.

**Comment by soumya997** [comment]
plz, give some response if it's ok, and I will push the changes.

**Comment by Nibba2018** [comment]
@soumya997 Try to show the commits value.

**Comment by soumya997** [comment]
> @soumya997 Try to show the commits value.

<a href="https://ibb.co/sQpy9z8">Screenshot-648</a>

Is it ok now @Nibba2018 ?

**Comment by Nibba2018** [comment]
Yeah better @soumya997

**Comment by soumya997** [comment]
Should I push from the same branch or I should create a different one?
If the same branch these will be the steps(after switching to this branch)
I have to do `git add .`, `git commit -m "<message>" ` and `git push -u origin <branch-name>` , right @Nibba2018 ?

**Comment by Nibba2018** [comment]
@soumya997 just do git push after git commit.

**Comment by soumya997** [comment]
@Nibba2018 I just pushed the code, please check.
@Nibba2018 @BALaka-18 @khanfarhan10 I hope all the issues are fixed now. It was so much learning for me specifically building the web app and pushing the code using Git(as I had less experience with git). Thank you for helping me out🙇‍♂️.

**Comment by soumya997** [comment]
Was my code readable?

File: README.md
Comment (Nibba2018) [inline README.md]: Make the `web-app` text clickable by `[web-app](link_to_webapp)` and remove `Link of the website...`

File: app.py
Comment (Nibba2018) [inline app.py]: `for` February enter 2

File: app.py
Comment (Nibba2018) [inline app.py]: Dont provide ranges. Just provide an example value.

File: app.py
Comment (Nibba2018) [inline app.py:431]: Looks like you are hard coding the information here. We want it to be dynamic so that new changes are visible as well.

**Review by Nibba2018 — CHANGES_REQUESTED** [review]
@soumya997 I have added some comments below.

File: README.md
Comment (soumya997) [reply README.md]: Ok I will fix that

File: app.py
Comment (soumya997) [reply app.py:54]: You should be saying that when I asked you checked those screenshots. Just make it at once.
Every time I am pushing the code I am asking you to check each and every feature I am adding. And you were saying yes its ok. Now why you are turning now, the link of the deployed app is always the same.

File: app.py
Comment (soumya997) [reply app.py]: I am not complaining but as a maintainer, you should be careful, every time I am adding any features and I was also showing that in the form of screenshots then I was asking for "should I push the changes?".every time you were saying yes ok push the changes.
So why to ask for changes now.
Obviously, I will change what you have asked for, but make sure these are the last changes.

File: app.py
Comment (soumya997) [reply app.py:431]: yes, it's hard coded, please show me some tutorial as I have very less experience of doing things dynamically,
The things you are seeing, it took me 2 to 3 continuous hours to figure that out how to add those red and green texts, as everything was getting spread out.
Or you may go for a slightly experienced person to handle that.

File: app.py
Comment (Nibba2018) [reply app.py]: @soumya997 How am I supposed to know if your code is static or dynamic based on your screenshots? If you dont push your changes I can't see what code you have written.

Comment (soumya997) [reply app.py]: And the range was mentioned in the readme, Value of those parameters ranges between that and if you give more than or less than the values of range it will not make any sense, not for the model(as for the model it would be new data and made predictions would be wrong) and not in general.
So, review the above statement👆 and say you still want to see some changes or not.

Comment (soumya997) [reply app.py]: Adding a place holder is static, and you can easily see that via a screenshot.
When it's about the dynamic feature that you requested for the contributors page for that you can say that you cant see via a screenshot, and for that I have handed over to you, see comment below.

Comment (Nibba2018) [reply app.py]: > And the range was mentioned in the readme, Value of those parameters ranges between that and if you give more than or less than the values of range it will not make any sense,

Looks like you haven't researched about the features. The values provided in the readme were the ranges within the dataset and not the actual parameter ends for that feature. for e.g

* FFMC ranges from 0-101 and not 18.7 to 96.20
* DMC is open ended which means it can have any value and it not closed between 1.1 to 291.3
* DC ranges from 0 - 1000 and not in between 7.9 to 860.0
* ISI is also open ended.
* Same goes for temp, wind, rain etc.

> not for the model(as for the model it would be new data and made predictions would be wrong) and not in general.

Well in that case the model is overfitting and should not be used in the first place.

File: app.py
Comment (Nibba2018) [reply app.py:431]: > please show me some tutorial as I have very less experience of doing things dynamically

That's why I asked you to refer [this](https://github.com/fury-gl/fury/blob/master/docs/source/ext/github_tools.py) for this reason.

@soumya997 Okay this can be done on a separate PR.

File: app.py
Comment (soumya997) [reply app.py]: > Well in that case the model is overfitting and should not be used in the first place.

Its a good point but I said that on mode retraining perspective, do you know that after deployment there is a concept called model retraining, retraining of the model happens when the model starts to perform worse every time and it performs worse because it needs to predict the data which all new to it.
And except that and I am also not complaining by saying that but the csv file was of 500 rows.

Comment (soumya997) [reply app.py]: > Looks like you haven't researched about the features. The values provided in the readme were the ranges within the dataset and not the actual parameter ends for that feature. for e.g

Thanks for pointing that out yes I did not and I am sorry for that.

Comment (soumya997) [reply app.py]: Ok @Nibba2018 , I am pushing the changes except for the contributors commit feature.

File: app.py
Comment (soumya997) [reply app.py:431]: I will try it soon, but for now, it will be good for me to skip that part. Thank you for suggesting that @Nibba2018 .

Comment (soumya997) [reply app.py:431]: I tried some StackOverflow code too but did not work for me. This was the code, take a look if time permits. I don't vate the link of the thread but I have the code👇
```from requests import session
def login():
    sess = session()

    # login here and return session with valid creds
    return sess

def generateList(link):
    # you need to login before you do anything
    sess = login()

    # because of the way that requests works, you must start out by creating an object to
    # imitate the response object. This will help you to cleanly while-loop through
    # github's pagination
    class response_immitator:
        links = {'next': {'url':link}}
    response = response_immitator() 
    while 'next' in response.links:
        response = sess.get(response.links['next']['url'])
        for repo in response.json():
            yield repo

def check_commit_count(baseurl, user_name, repo_name, max_commit_count=None):
    # login first
    sess = login()
    if max_commit_count != None:
        totalcommits = 0

        # construct url to paginate
        url = baseurl+"repos/" + user_name + '/' + repo_name + "/stats/contributors"
        for stats in generateList(url):
            totalcommits+=stats['total']

        if totalcommits >= max_commit_count:
            return False
        else:
            return True

def main():
    # what user do you want to check for commits
    user_name = "soumya997"

    # what repo do you want to check for commits
    repo_name = "AI-Hacktoberfest"

    # github's base api url
    baseurl = "https://api.github.com/"

    # call function
    check_commit_count(baseurl, user_name, repo_name,30)

if __name__ == "__main__":
    main()
```

File: app.py
Comment (soumya997) [reply app.py]:
```
x = st.text_input("X","Enter an integer e.g. 5")
y = st.text_input("Y","Enter an integer e.g. 5")
```

This is the code for variable "X" and "Y", should I change the placeholders too like: `e.g. 5`, or should I let it be?

Comment (Nibba2018) [reply app.py]: > And except that and I am also not complaining by saying that but the csv file was of 500 rows.
I know, we are not "actually" deploying a model so it alright. :laughing: . Mentioning such a range is like asking your teacher to ask viva questions from the only chapter that you have studied. :laughing: 

> Its a good point but I said that on mode retraining perspective, do you know that after deployment there is a concept called model retraining, retraining of the model happens when the model starts to perform worse every time and it performs worse because it needs to predict the data which all new to it.

Yeah thats when we want to retrain the model so that it can learn new features from the new data but in this case we only want it to predict.

> This is the code for variable "X" and "Y", should I change the placeholders too like: e.g. 5, or should I let it be?

This is fine, no need to change this.

>  Ok @Nibba2018 , I am pushing the changes except for the contributors commit feature.

LGTM! :rocket:

File: app.py
Comment (Nibba2018) [reply app.py:431]: @soumya997 , No problem we will create a separate issue for this. There's already a lot of work being done in this PR.

**Review by Nibba2018 — APPROVED** [review]

File: app.py
Comment (soumya997) [reply app.py]: @Nibba2018 I have pushed the changes, please check.

**Comment by Nibba2018** [comment]
@soumya997 , Thank you these PR. Merging :guitar:

File: app.py
Comment (soumya997) [reply app.py]: > I know, we are not "actually" deploying a model so it alright. 😆 . Mentioning such a range is like asking your teacher to ask viva questions from the only chapter that you have studied. 😆


If you remember properly I did use the whole 500 data samples for training, only 80% was for training, so the rest of the data was for validation. So the whole range of each feature (like- for FFMC- 18.7 to 96.20) was not going training. 
There I just tried to mention what was provided in the readme.
And questions are always out of the syllabus.

**Comment by soumya997** [comment]
> @soumya997 , Thank you these PR. Merging 🎸

Thank you @Nibba2018 for always helping me out with git commands and reviewing the changes.

**Pull request merged**
